graph-api.yaml

```diff
@@ -0,0 +1,459 @@
+        503:
+          description: Service Unavailable, i.e. the server is still in the process of starting.
+  /{dataset}/update:
+    post:
+      summary: Execute a SPARQL Update
+      description: | 
+        Submit a SPARQL update against a dataset and, by means of the RDF Patches generated, also update the other Smart Caches.
+      parameters:
+        - $ref: '#/components/parameters/dataset'
+      requestBody:
+        required: true
+        content:
+          application/sparql-update:
+            schema:
+              type: string
+              description: SPARQL Update query string.
+      responses:
+        200:
+          description: Update executed successfully.
+        400:
+          description: Invalid query or bad request.
+        401:
+          description: Unauthorised query.
+        404:
+          description: Endpoint not found i.e. incorrect dataset.
+        500:
+          description: Internal Server Error i.e. the server failed to process the request.
+        503:
+          description: Service Unavailable, i.e. the server is still in the process of starting.
+  /{dataset}/shacl:
+    get:
+      summary: Retrieve SHACL shapes for a dataset
+      description: |
+        Retrieve the SHACL validation shapes associated with a specific dataset.
+        The shapes are returned in RDF format and can be used to validate data against the dataset's SHACL rules.
+      parameters:
+        - $ref: '#/components/parameters/dataset'
+      responses:
+        200:
+          description: SHACL shapes retrieved successfully.
+          content:
+            application/turtle:
+              schema:
+                type: string
+                description: SHACL shapes in Turtle format.
+            application/ld+json:
+              schema:
+                type: string
+                description: SHACL shapes in JSON-LD format.
+            application/n-triples:
+              schema:
+                type: string
+                description: SHACL shapes in N-Triples format.
+        404:
+          description: SHACL shapes not found for the specified dataset.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        500:
+          description: Internal Server Error.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+    post:
+      summary: Validate RDF data against SHACL shapes
+      description: |
+        Submit RDF data and SHACL shapes to validate the data. 
+        The server processes the request and returns a SHACL validation report.
+      parameters:
+        - $ref: '#/components/parameters/dataset'
+      requestBody:
+        description: RDF data and SHACL shapes to validate.
+        required: true
+        content:
+          application/ld+json:
```

Comment:
Although [the spec](https://www.w3.org/TR/shacl/) says you can use JSON-LD and other formats, I don't see any support in Fuseki for that - only for Turtle.